pilot/pkg/features/pilot.go

```diff
@@ -163,6 +163,12 @@ var (
 			"and will be removed in the near future.",
 	)
 
+	EnableProtocolSniffing = env.RegisterBoolVar(
+		"PILOT_ENABLE_PROTOCOL_SNIFFING",
+		false,
+		"If enabled, protocol sniffing will be used on ports whose port protocol is not specified or unsupported",
```

Comment:
I wouldn't use 'unsupported' - unknown maybe ?

Reply:
+1, more details would be useful. Indicate that it can detect http vs tcp automatically, not things like mongo. mention any tradeoffs, unsupported parts, etc